canvas/biplot.py

```diff
@@ -0,0 +1,499 @@
+                colorVal.append(scalarMap.to_rgba(index))
+
+            if show_color_drop is False:
+                sample_color_dict = dict(zip(sample_group_append, colorVal))
+                sample_color_dict = OrderedDict(sorted(sample_color_dict
+                                                       .items(),
+                                                       key=lambda x: x[0],
+                                                       reverse=True))
+
+                for name, group in
+                sample_df.groupby(sample_df_metadata[sample_color_category]):
+                    ax.plot(np.ravel(group[sample_df_x]),
+                            np.ravel(group[sample_df_y]),
+                            marker=samp_marker, linestyle='', ms=samp_ms,
+                            color=sample_color_dict[name],
+                            label=name, alpha=samp_alpha)
+
+            if show_color_drop is True:
+
+                color_drop_append = []
+                if len(sample_drop_col) == 1:
+                    for index in range(len(sample_drop_list)):
+                        color_drop_append.append(sample_drop_col[0])
+                    colorVal = colorVal + color_drop_append
+
+                if len(sample_drop_col) == len(sample_drop_list):
+                    for index in range(len(sample_drop_list)):
+                        color_drop_append.append(sample_drop_col[index])
+                    colorVal = colorVal + color_drop_append
+
+                sample_group_append = list(sample_group_append) +
+                list(sample_drop_list)
+                sample_color_dict = dict(zip(sample_group_append, colorVal))
+                sample_color_dict = OrderedDict(sorted(sample_color_dict
+                                                       .items(),
+                                                       key=lambda x: x[0],
+                                                       reverse=True))
+
+                for name, group in sample_df_copy
+                .groupby(sample_df_metadata_copy[sample_color_category]):
+                    if name not in sample_drop_list:
+                        ax.plot(np.ravel(group[sample_df_x]),
+                                np.ravel(group[sample_df_y]),
+                                marker=samp_marker, linestyle='', ms=samp_ms,
+                                color=sample_color_dict[name],
+                                label=name, alpha=samp_alpha)
+
+                for name, group in sample_df_copy
+                .groupby(sample_df_metadata_copy[sample_color_category]):
+                    if name in sample_drop_list:
+                        ax.plot(np.ravel(group[sample_df_x]),
+                                np.ravel(group[sample_df_y]),
+                                marker=samp_marker, linestyle='', ms=samp_ms,
+                                color=sample_color_dict[name],
+                                label=name, alpha=samp_alpha)
```

Comment:
this is a bit code that's supposed to plot and color "dropped" and put them in the bottom of the legend